synapse/handlers/e2e_keys.py

```diff
@@ -59,6 +59,9 @@ def __init__(self, hs: "HomeServer"):
 
         federation_registry = hs.get_federation_registry()
 
+        self._allow_device_name_lookup_over_federation = (
+            hs.config.federation.allow_device_name_lookup_over_federation
+        )
```

Comment:
As an aside: I've never understood why we copy everything off the config objects. It makes it much harder to dynamically reload config, and only saves a few getattr calls and indirections---is reading from the config really that much of a bottleneck?

Reply:
It was much more of a concern before https://github.com/matrix-org/synapse/pull/10985. Nowadays perhaps not so. You're right in that it adds clutter and makes reloading config more difficult. I think I'll remove the class variable for now, but it'd be good to have a team discussion on whether we should stick to one way or the other.

Reply:
(I don't particularly mind leave this as written here---I'm just thinking out loud.)

Reply:
Changed in https://github.com/matrix-org/synapse/pull/14304/commits/e1614de7b95c6ebd64d542a6666b85a3c64f02d1. One thing it did require is a `self.config = config` line (or alternatively a `self.hs = hs`) in `__init__`. I wonder if this has any performance impact?

Reply:
Also discussed at https://element.topicbox.com/groups/backend/T1a142202e6ba4531-M8be7fbe3edd34e232ebb168e